taskipy/pyproject.py

```diff
@@ -10,29 +10,47 @@
     MissingPyProjectFileError,
     MissingTaskipyTasksSectionError
 )
+from taskipy.variable import Variable
 
 
 class PyProject:
     def __init__(self, base_dir: Path):
         pyproject_path = self.__find_pyproject_path(base_dir)
-        self.__items = PyProject.__load_toml_file(pyproject_path)
+        self.__items = self.__load_toml_file(pyproject_path)
 
     @property
     def tasks(self) -> Dict[str, Task]:
         try:
-            return {
-                task_name: Task(task_name, task_toml_contents) for
-                task_name, task_toml_contents in self.__items['tool']['taskipy']['tasks'].items() }
+            toml_tasks = self.__items['tool']['taskipy']['tasks'].items()
         except KeyError:
             raise MissingTaskipyTasksSectionError()
 
+        tasks = {}
+        for name, toml_contents in toml_tasks:
+            tasks[name] = Task(name, toml_contents)
+
+        return tasks
+
     @property
-    def variables(self) -> Dict[str, Task]:
+    def variables(self) -> Dict[str, Variable]:
         try:
-            return self.__items['tool']['taskipy'].get('variables', {})
+            toml_vars = self.__items['tool']['taskipy'].get('variables', {})
         except KeyError:
             return {}
 
+        vars_dict = {}
+        for name, toml_contents in toml_vars.items():
+            if isinstance(toml_contents, str):
+                vars_dict[name] = Variable(name, toml_contents, recursive=False)
+            elif toml_contents.get('var') or toml_contents.get('recursive'):
```

Comment:
Having `var` is a must, but having `recursive` is not a must. Therefore, the condition should be:
1. This is a dict
2. It has `var` and `var` is a string.

```suggestion
            elif isinstance(toml_contents, dict) and isinstance(toml_contents.get('var'), str):
```

We don't need to check for `recursive` in the condition, because it's ok - we fallback to `False` if it doesn't exist (line 49).

Finally, we shouldn't allow variables that are not string or dicts that have {var}. So we need to have one final else, that throws if any of the variables is malformed:

```python
else:
    raise InvalidVariableError(name)
```

And also test that :)